src/main/scala/org/broadinstitute/hail/sparkextras/OrderedPartitioner.scala

```diff
@@ -58,43 +56,55 @@ case class OrderedPartitioner[PK, K](rangeBounds: Array[PK], numPartitions: Int,
         partition = rangeBounds.length
       }
     }
-    if (ascending) {
-      partition
-    } else {
-      rangeBounds.length - partition
-    }
+    partition
   }
 
   override def equals(other: Any): Boolean = other match {
-    case r: OrderedPartitioner[_, _] => r.rangeBounds.sameElements(rangeBounds) && r.ascending == ascending
+    case r: OrderedPartitioner[_, _] => r.rangeBounds.sameElements(rangeBounds) && r.numPartitions == numPartitions
```

Comment:
this should check that the ordered key for `this` is equal to the ordered key for `other`.  This was an oversight in the current master and I couldn't figure out how to deal with the type erasure issue until realizing we need to check the implicit keys too.